Configure ArcGIS Server to use Active Directory

mu88 opened the pull request:
This commit enables the Cookbook to set the ArcGIS Server's identity store to an Active Directory. For the authentication against the Active Directory, the following parameters out of `webgis-windows.json` are used:

- `node['arcgis']['run_as_user']`
- `node['arcgis']['run_as_password']`

To enable the coupling, the parameter `node['arcgis']['server']['configure_active_directory']` within `webgis-windows.json` has to be set to `true`.

Furthermore, the ArcGIS Server's administrators and publishers can be bound to Active Directory groups. For this, the parameters `node['arcgis']['server']['active_directory_groups_administer']` and `node['arcgis']['server']['active_directory_groups_publisher']` have to be set within `webgis-windows.json`. These parameters are arrays, so several Active Directory groups can be set as administrators or publishers.

The new functionality is encapsulated within the recipe `server_active_directory` which is not enabled by default within 'webgis-windows.json' - this can be done on demand.

Commit:
Configure ArcGIS Server to use Active Directory

kirankc commented:
@mu88, this is very helpful. I hope ESRI accepts this commit and merges it to the base as a lot of organizations could benefit from this.

pbobov reviewed (commented):
This functionality (with a few changes to make it more generic) will be available in the next release of arcgis-enterprise cookbook.

kirankc commented:
@pbobov that's good to hear. Any rough estimate on when the next release would be? Also, I am assuming you are referring to 10.5.1??

pbobov commented:
@kirankc Yes. The plan is to release ArcGIS Chef Cookbooks 3.1.0 with release of ArcGIS 10.5.1.

mu88 commented:
@kirankc Great to hear that it would help you. Just keep in mind that my enhancement only supports ArcGIS Server's basic Active Directory support, so no nested Active Directory groups are supported. For this, I've also implemented Esri's ASP.NET Provider within Chef.
So if you're interested in this or other things (I've also implemented stuff like registering data sources (SDE and folders) in ArcGIS Server or create backups of ArcGIS Server / Portal)  please let me know and I'll try to provide another PR to Pavel.

kirankc commented:
@mu88 would love to see them. Please share the goodies with the community and @pbobov  - I am particularly interested in the backup pieces. And, thank you for the heads up on nested AD groups - curious to see what ESRI has in cookbooks 3.1.0 towards this.

mu88 commented:
Good morning everybody,
I've provided all my enhancements:

- [Provide the ability to maintain data sources](https://github.com/Esri/arcgis-cookbook/pull/81)
- [Provide extended functionality to authenticate with Portal](https://github.com/Esri/arcgis-cookbook/pull/80)
- [Install ArcGIS Patches](https://github.com/Esri/arcgis-cookbook/pull/79)
- [Set SOC Maximum Heap Size](https://github.com/Esri/arcgis-cookbook/pull/78)
- [Create backups](https://github.com/Esri/arcgis-cookbook/pull/77)
- [Use ASP.NET Provider to couple ArcGIS Server and Active Directory](https://github.com/Esri/arcgis-cookbook/pull/76)

I hope you find it useful!
Regards and have a nice day =)

kirankc commented:
Thank you @mu88 for sharing your hard work with the community. This is good stuff and something we all could benefit from.
